tests/api/show-spec.js

```diff
@@ -0,0 +1,94 @@
+var frisby = require('frisby');
+var url = require('./lib/url');
+
+frisby.globalSetup({
+  request: {
+    headers: {'Accept': 'application/json'},
+    inspectOnFailure: true
+  }
+});
+
+var user = {
+  fname: 'Travis',
+  sname: 'The Tester',
+  email: 'showspecdevnull' + Date.now() + '@example.com',
+  receive_email: false
+};
+
+var show = {
+  title: 'Travis likes to test shows.',
+  description: 'Travis is a busy guy, but in his spare time, he does his own radio show to test what this on the air malarkey is all about.',
+};
+
+frisby.create('Create a test member')
+  .post(url.base + 'user?api_key=travis-test-key', user, {json: true})
+  .expectStatus(201)
+  .afterJSON(function(json) {
+    var memberid = json.payload.memberid;
+    show.credits = [{type: 1, memberid: memberid}];
+
+    frisby.create('Create a test show')
+      .post(url.base + 'show?api_key=travis-test-key', show, {json: true})
+      .expectStatus(201)
+      .expectHeaderContains('content-type', 'application/json')
+      .expectJSON({
+        status: 'OK',
+        payload: {
+          title: show.title,
+          description: show.description
+        }
+      })
+      .expectJSONTypes({
+        time: String,
+        payload: {
+          show_id: Number,
+          // Don't like, but backwards compatibility :(
+          credits: String
+        }
+      })
+      .afterJSON(function(json) {
+        var showid = json.payload.show_id;
+
+        frisby.create('The show should have no seasons')
+          .get(url.base + 'show/' + showid + '/numberofseasons?api_key=travis-test-key')
+          .expectStatus(200)
+          .expectHeaderContains('content-type', 'application/json')
+          .expectJSON({
+            status: 'OK',
+            payload: 0
+          })
+          .expectJSONTypes({
+            time: String
+          })
+          .toss();
+
+        frisby.create('The show should have a credit')
+          .get(url.base + 'show/' + showid + '/credits?api_key=travis-test-key')
+          .expectStatus(200)
+          .expectHeaderContains('content-type', 'application/json')
+          .expectJSON({
+            status: 'OK',
+            payload: show.credits
+          })
+          .expectJSONTypes({
+            time: String
+          })
+          .toss();
+
+        frisby.create('The show should appear in the All Shows list')
+          .get(url.base + 'show/allshows?api_key=travis-test-key')
+          .expectStatus(200)
+          .expectHeaderContains('content-type', 'application/json')
+          .expectJSON('payload.?', {
+            show_id: showid,
+            title: show.title,
+            description: show.description
+          })
+          .expectJSONTypes({
+            time: String
+          })
+          .toss();
```

Comment:
I know it's sort of JS's thing, but are test cases inside test cases a good idea? I think I'd prefer to see them separated out. Removes the need for this "toss() toss() toss()" silliness :)